tutorials/graph-theory/binary-search-tree.md

```diff
@@ -1,12 +1,391 @@
 ---
-title: 'Binary Search Tree'
-description: 'Coming Soon!'
+title: "Binary Search Tree (BST)"
+description: "Author: @utkarsh"
 draft: true
+hide_table_of_contents: true
 keywords:
   - leetcode
   - tutorial
   - binary search tree
   - data structure
 ---
 
-Coming Soon
\ No newline at end of file
+<TutorialAuthors names="@utkarsh"/>
+
+## Overview
+
+Before jump to BST. We recap some basics to better understand BST:
+
+A tree is a data structure composed of nodes that has some following characteristics:
+
+1. Each tree has root node at the top (also known as Parent Node).
+2. The root node has zero or more child nodes.
+3. Each child node has zero or more child nodes. This create subtree in the tree. Every node has its own subtree made up of its children etc. This means that every node on its own can be tree.
```

Comment:
i believe these are copied from https://www.freecodecamp.org/news/binary-search-trees-bst-explained-with-examples/

Reply:
yes, you want to remove it?

Reply:
ok

Reply:
Added reference please check that